src/components/RestaurantList/RestaurantList.tsx

```diff
@@ -0,0 +1,25 @@
+import styles from './style.module.css';
+import { memo } from 'react';
+import { Restaurant } from '../../types';
+import RestaurantItem from '../RestaurantItem/RestaurantItem';
+
+interface RestaurantListProps {
+  restaurantList: Restaurant[];
+  onItemClick: (restaurant: Restaurant) => void;
+}
+
+function RestaurantList({ restaurantList, onItemClick }: RestaurantListProps) {
+  const createRestaurantItemElements = () => {
+    return restaurantList.map((restaurant) => (
+      <RestaurantItem key={restaurant.id} restaurant={restaurant} onClick={onItemClick} />
+    ));
+  };
+
+  return (
+    <section className={styles.restaurantListContainer}>
+      <ul className="restaurant-list">{createRestaurantItemElements()}</ul>
+    </section>
+  );
```

Comment:
```suggestion
  return (
    <section className={styles.restaurantListContainer}>
      <ul className="restaurant-list">
        {restaurantList.map((restaurant) => (
          <RestaurantItem
            key={restaurant.id}
            restaurant={restaurant}
            onClick={onItemClick}
          />
        ))}
      </ul>
    </section>
  );
```

jsx에서 뷰를 한번에 볼 수 있게 해주고, 

onItemClick과 같은 로직이 담긴 것들을 별도로 함수로 선언하는게, 

일반적입니다~!